doc/docusaurus/docs/delve-deeper/languages.md

```diff
@@ -0,0 +1,100 @@
+
+Second, users will need to adopt a new programming language and incorporate it into their existing tech stacks.
+This can present a considerable challenge, as it involves a learning curve, increased cognitive load, and the necessity to introduce and manage additional tools.
+
+## Embedded DSLs
+
+An embedded DSL (commonly referred to as an eDSL) generally takes the form of a library in a host programming language.
+Functional languages such as Haskell are particularly well-suited for hosting eDSLs, as the implementation of an eDSL largely involves functions that construct and transform abstract syntax trees (ASTs).
+
+The disadvantages of standalone DSLs are, in contrast, the advantages of embedded DSLs.
+Embedded DSLs, however, come with the drawback that the complexity of constructing and manipulating ASTs are exposed to the users.
+When using an embedded DSL, you are essentially writing programs that create and manage ASTs, rather than straightforward code.
+
+Take, for instance, a program that accepts two integers as input, and checks if the first is less than the second.
+Normally, you would write a function of type `Integer -> Integer -> Bool`, which takes two integers and returns a boolean.
+However, when working with an eDSL, your program might have a type like `AST Integer -> AST Integer -> AST Bool`, which takes two ASTs that evaluate to integers, combines them, and yields a larger AST that evaluates to a boolean.
+Like standalone DSLs, this also introduces additional learning curves and cognitive load, though for a different reason.
+
+Another disadvantage of eDSLs is that it is harder, compared to the other two approaches, to produce readable target code or accurate source mappings for debuggers.
+This stems from the nature of eDSLs, which are libraries that construct and manipulate ASTs.
+Since they do not have direct access to the host language's ASTs, it can be challenging to retrieve information related to the source code, such as variable names, module names and code locations.
+
+There's another category of eDSLs, known as "shallow embedding", which, unlikely the "deep embedding" described earlier, does not construct intermediate ASTs.
+Instead, shallow embedding involves using overloaded functions.
+For example, a DSL designed as a shallow embedding for working with databases might include operations such as `createTable`, `getItem`, and `putItem`.
+These functions are overloaded, allowing them to work with various database implementations, including mock databases for testing purposes.
+Such overloaded functions are typically defined using typeclasses in functional languages, or interfaces/traits in object-oriented languages.
+
+While it is valid to call shallow embeddings _languages_, it is a bit of a stretch.
+Overloaded functions are widespread in everyday programming, and are not usually regarded as languages due to the absence of ASTs.
+All existing eDSLs targeting UPLC are examples of deep embeddings.
```

Comment:
Would it even make any sense to have a shallowly embedded language that targets UPLC?

Reply:
Great question. When targeting UPLC, since the whole point is to construct an AST, a shallow embedding would make much less sense. I'll clarify it in the doc.

Reply:
Added "Moreover, shallow embedding is less fitting when the eDSL targets a lower level language like UPLC, as constructing ASTs for UPLC will still be necessary."